src/app/data-analytics-dashboard/oncology/oncology-reports/oncology-report-pdf-view/oncology-report-pdf.service.ts

```diff
@@ -0,0 +1,502 @@
+              {
+                text: this._formatReportIndicators(title) + ' Aggregate Report',
+                style: 'header',
+                alignment: 'center'
+              },
+              {
+                columns:
+                  [
+                    { width: '*', style: 'dateColumn', text: 'Start Date: ' + params.startDate },
+                    { width: '*', style: 'dateColumn', text: 'End Date: ' + params.endDate }
+                  ],
+                bold: true,
+                color: '#2a2a2a',
+                fillColor: '#d3d3d3'
+              },
+              aggregatedData.sections[0].body,
+            ]
+          },
+          {
+            layout: 'noBorders',
+            table: {
+              widths: ['*', '*'],
+              body: this.constructAggregateOuterLayout(data, params)
+            }
+          }
+          ],
+          styles: {
+            header: {
+              fontSize: 14,
+              bold: true,
+              margin: [0, 0, 0, 10]
+            },
+            locationStyle: {
+              fontSize: 12,
+              bold: true,
+              margin: [0, 10, 0, 5]
+            },
+            dateColumn: {
+              alignment: 'center',
+              fontSize: 10,
+              bold: true,
+            },
+            defaultTable: {
+              fontSize: 8,
+              margin: [0, 10, 0, 5]
+            },
+            headerstyle: {
+              fontSize: 8,
+              bold: true,
+              color: '#2a2a2a',
+              fillColor: '#d3d3d3'
+            },
+            aggTable: {
+              fontSize: 8,
+              margin: [0, 10, 0, 5]
+            }
+          },
+          defaultStyle: {
+            fontSize: 8
+          }
+        });
+      });
+    }).pipe(first());
+
+  }
+
+  public constructPdfSections(data: Array<any>, params: any): Array<Array<any>> {
+    const sectionsArray: Array<Array<any>> = [[]];
+    let section: Array<any> = [];
+    data.forEach(rowSection => {
+      section = [
+        {
+          style: 'subheader',
+          table: {
+            widths: ['*', 'auto'],
+            body: [
+              [
+                { text: 'Facility Name: ' + rowSection.location_name, style: 'headerStyle' },
+                { text: 'Date: ' + rowSection.encounter_datetime, style: 'headerStyle' }
+              ]
+            ]
+          }
+        },
+        this.constructBodySection(rowSection, params, 'pdf')
+
+      ];
+
+      sectionsArray.push(section);
+    });
+
+    return sectionsArray;
+  }
+
+  public constructAggregateOuterLayout(data: Array<any>, params: any): Array<Array<Array<any>>> {
+    const sectionsArray: Array<Array<Array<any>>> = [[[]]];
+
+    const aggregatedData = this.aggregateData(data, params);
+
+    aggregatedData.sections.shift();
+
+    const left_temp: Array<Array<any>> = [[]];
+    const right_temp: Array<Array<any>> = [[]];
+    aggregatedData.sections.forEach((arr) => {
```

Comment:
Considering that you are looping through sections, it would make sense if you contextualize that in the name of param **arr**. Use **section** instead of **arr**. It would be nice if you did this in the whole of the commit.